src/components/CreateMsa.svelte

```diff
@@ -1,27 +1,36 @@
 <script lang="ts">
   import type { web3Enable, web3FromSource } from '@polkadot/extension-dapp';
   import { dotApi } from '$lib/stores';
-  import type { DotApi } from '$lib/storeTypes';
+  import type { DotApi, MsaInfo } from '$lib/storeTypes';
   import { defaultDotApi } from '$lib/storeTypes';
   import type { ApiPromise } from '@polkadot/api';
   import { isLocalhost } from '$lib/utils';
-  import { submitCreateMsa } from '$lib/connections';
+  import { submitCreateMsa, type TxnFinishedCallback } from '$lib/connections';
   import TransactionStatus from '$components/TransactionStatus.svelte';
   import { isFunction } from '@polkadot/util';
   import { onMount } from 'svelte';
+  import { user } from '$lib/stores/userStore';
+  import { getMsaInfo } from '$lib/polkadotApi';
+  import { pageContent } from '$lib/stores/pageContentStore';
 
   let localDotApi: DotApi = defaultDotApi;
   let thisWeb3FromSource: typeof web3FromSource;
   let thisWeb3Enable: typeof web3Enable;
   let showTransactionStatus = false;
   export let txnStatuses: Array<string> = [];
-  export let validAccounts = {};
-  export let signingAddress = '';
   // a callback for when the user cancels this action
-  export let cancelAction = () => {};
+  export let cancelAction = () => {
+    pageContent.login();
+  };
+
   // a callback for when a transaction hits a final state
-  export let txnFinished = () => {
+  export let txnFinished: TxnFinishedCallback = async (succeeded: boolean) => {
```

Comment:
Nit: for booleans like this, I typically like to use the prefix "is" so that the type and if's are very clear. In this case: "isSuccess"